File: McbEdu.Mentorias.ShopDemo.Services/Products/Interfaces/IProductService.cs
using McbEdu.Mentorias.DesignPatterns.NotificationPattern;
using McbEdu.Mentorias.ShopDemo.Domain.Contexts.ProductContext.DTO;
using McbEdu.Mentorias.ShopDemo.Services.Products.Inputs;

namespace McbEdu.Mentorias.ShopDemo.Services.Products.Interfaces;

public interface IProductService
{
    Task<(bool, List<NotificationItem>)> ImportProductAsync(ImportProductServiceInput input);
    Task<(bool HasExecuted, List<NotificationItem> Notifications, List<Product> Products)> GetProductsAsync(GetProductServiceInput input);
}

File: src/main/java/com/mcbmentorias/shopdemo/domain/services/interfaces/IProductService.java
package com.mcbmentorias.shopdemo.domain.services.interfaces;

import com.mcbmentorias.shopdemo.domain.entities.product.Product;
import com.mcbmentorias.shopdemo.domain.entities.product.inputs.CreateNewImportProductInput;

import java.util.Collection;
import java.util.List;

public interface IProductService {

    Boolean importProduct(
            final CreateNewImportProductInput input
    );

    Boolean createImportProductOrNotifyDifferences(final CreateNewImportProductInput domainInput);

    Product getByCode(final String code);

    Boolean createImportProductOrNotifyDifferences(final Collection<CreateNewImportProductInput> inputs);
}
